src/main/java/org/wso2/lsp4intellij/contributors/rename/LSPRenameHandler.java

```diff
@@ -88,9 +91,11 @@ private InplaceRefactoring doRename(PsiElement elementToRename, Editor editor) {
                         public void pass(PsiElement element) {
                             MemberInplaceRenamer renamer = createMemberRenamer(element,
                                     (PsiNameIdentifierOwner) elementToRename, editor);
-                            boolean startedRename = renamer.performInplaceRename();
-                            if (!startedRename) {
-                                performDialogRename(editor);
+                            if (isInplaceRename) {
+                                renamer.performInplaceRename();
+                            } else {
+                                isInplaceRename = true;
```

Comment:
Why are we setting `isInplaceRename` as true in here?

Reply:
It is set to true to reset the state, ensuring that the next rename request checks again whether it involves an in-place rename or a multiple file rename.